DataCapturing/DataCapturingService.swift

```diff
@@ -368,31 +374,46 @@ extension DataCapturingService: CLLocationManagerDelegate {
     /**
      The listener method that is informed about new geo locations.
 
+     - Remark:
+        This function is one of the most critical parts of the `DataCapturingService`. It is called once per second and should not do any unncessary work.
      - Parameters:
-     - manager: The location manager used.
-     - didUpdateLocation: An array of the updated locations.
+        - manager: The location manager used.
+        - didUpdateLocation: An array of the updated locations.
      */
     public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
 
         for location in locations {
             // Smooth the way by removing outlier coordinates.
             let howRecent = location.timestamp.timeIntervalSinceNow
             guard location.horizontalAccuracy < 20 && abs(howRecent) < 10 else { continue }
+            let timestamp = DataCapturingService.convertToUtcTimestamp(date: location.timestamp)
 
             let geoLocation = GeoLocation(
                 latitude: location.coordinate.latitude,
                 longitude: location.coordinate.longitude,
                 accuracy: location.horizontalAccuracy,
                 speed: location.speed,
-                timestamp: DataCapturingService.convertToUtcTimestamp(date: location.timestamp))
+                timestamp: timestamp)
 
             lifecycleQueue.async(flags: .barrier) {
                 self.locationsCache.append(geoLocation)
             }
 
-            DispatchQueue.main.async {
-                self.handler(.geoLocationAcquired(position: geoLocation), .success)
+            geoLocationEventNumber += 1
+            if geoLocationEventNumber == 1 {
+                // TODO: Actually the calling app should care for whether this happens on the main thread or not. Nevertheless it must be synchronized. It could be added to the lifecycleQueue to achieve this. Maybe with version 5.
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self else {
+                        return
+                    }
+
+                    self.handler(.geoLocationAcquired(position: geoLocation), .success)
+                }
             }
+            if geoLocationEventNumber == locationUpdateSkipRate {
+                geoLocationEventNumber = 0
+            }
+
```

Comment:
this part was the core of the update.

Reply:
Maybe throw an IllegalArgumentException if someone defines a locationUpdateRate < 1;